### PR TITLE
Remove the DRF pin.

### PR DESCRIPTION
**Pull request by feanil**
Also test on all versions of DRF that we currently need to support.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,9 +58,9 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
 	pip-compile --upgrade -o requirements/travis.txt requirements/travis.in
 	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
-	# Let tox control the Django version for tests
-	sed '/^[d|D]jango==/d' requirements/test.txt > requirements/test.tmp
-	mv requirements/test.tmp requirements/test.txt
+	# Let tox control the Django and djangorestframework versions for tests
+	sed -i '/^[d|D]jango==/d' requirements/test.txt
```
Comment (nedbat) [inline Makefile]: Don't you need an empty-string argument for -i  ?

Comment (feanil) [reply Makefile]: @nedbat good catch, on linux the argument is optional, but on Macs it is required.  I've updated the PR to work with both.

**Comment by feanil** [comment]
@nedbat @ormsbee I think this is now ready for a final review.

File: Makefile [hunk 2]
```diff
@@ -58,9 +58,9 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	pip-compile --upgrade -o requirements/quality.txt requirements/quality.in
 	pip-compile --upgrade -o requirements/travis.txt requirements/travis.in
 	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
-	# Let tox control the Django version for tests
-	sed '/^[d|D]jango==/d' requirements/test.txt > requirements/test.tmp
-	mv requirements/test.tmp requirements/test.txt
+	# Let tox control the Django and djangorestframework versions for tests
+	sed -i"" '/^[d|D]jango==/d' requirements/test.txt
+	sed -i"" '/^djangorestframework==/d' requirements/test.txt
```
Comment (nedbat) [inline Makefile]: Need a space, or this is the same as no argument.

Comment (feanil) [reply Makefile]: Updated to work on both BSD and linux.

**Pull request merged**